# Data Exploration

***

## Selecting Columns

In [2]:
# Loading Python Libraries

import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

#### Loading Dataset

In [3]:
# Films Dataset

films = pd.read_csv("D:/documents/xlsx/films.csv")

#### Showing first few rows of the dataset

In [3]:
films.head()

id                                             title  release_year  \
0   1  Intolerance: Love's Struggle Throughout the Ages        1916.0   
1   2                    Over the Hill to the Poorhouse        1920.0   
2   3                                    The Big Parade        1925.0   
3   4                                        Metropolis        1927.0   
4   5                                     Pandora's Box        1929.0   

   country  duration language certification      gross     budget  
0      USA     123.0      NaN     Not Rated        NaN   385907.0  
1      USA     110.0      NaN           NaN  3000000.0   100000.0  
2      USA     151.0      NaN     Not Rated        NaN   245000.0  
3  Germany     145.0   German     Not Rated    26435.0  6000000.0  
4  Germany     110.0   German     Not Rated     9950.0        NaN

#### Checking the variables

In [5]:
films.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4968 entries, 0 to 4967
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             4968 non-null   int64  
 1   title          4968 non-null   object 
 2   release_year   4926 non-null   float64
 3   country        4966 non-null   object 
 4   duration       4955 non-null   float64
 5   language       4957 non-null   object 
 6   certification  4666 non-null   object 
 7   gross          4158 non-null   float64
 8   budget         4538 non-null   float64
dtypes: float64(4), int64(1), object(4)
memory usage: 349.4+ KB


#### Checking Missing Values

In [6]:
films.isna().any()

id               False
title            False
release_year      True
country           True
duration          True
language          True
certification     True
gross             True
budget            True
dtype: bool

**Missing Value Variables**:- release_year, country, duration, language, certification, gross, budget

In [7]:
# Select title column from table films

films[["title"]]

title
0     Intolerance: Love's Struggle Throughout the Ages
1                       Over the Hill to the Poorhouse
2                                       The Big Parade
3                                           Metropolis
4                                        Pandora's Box
...                                                ...
4963                                       Unforgotten
4964                                             Wings
4965                                        Wolf Creek
4966                                 Wuthering Heights
4967                           Yu-Gi-Oh! Duel Monsters

[4968 rows x 1 columns]

In [9]:
# Select release year from table
films.loc[:, ["release_year"]]

release_year
0           1916.0
1           1920.0
2           1925.0
3           1927.0
4           1929.0
...            ...
4963           NaN
4964           NaN
4965           NaN
4966           NaN
4967           NaN

[4968 rows x 1 columns]

In [10]:
# Select first 10 columns
films.head(10)

id                                             title  release_year  \
0   1  Intolerance: Love's Struggle Throughout the Ages        1916.0   
1   2                    Over the Hill to the Poorhouse        1920.0   
2   3                                    The Big Parade        1925.0   
3   4                                        Metropolis        1927.0   
4   5                                     Pandora's Box        1929.0   
5   6                               The Broadway Melody        1929.0   
6   7                                     Hell's Angels        1930.0   
7   8                                A Farewell to Arms        1932.0   
8   9                                       42nd Street        1933.0   
9  10                                She Done Him Wrong        1933.0   

   country  duration language certification      gross     budget  
0      USA     123.0      NaN     Not Rated        NaN   385907.0  
1      USA     110.0      NaN           NaN  3000000.0   100000.0  
2      USA     151.0      NaN     Not Rated        NaN   245000.0  
3  Germany     145.0   German     Not Rated    26435.0  6000000.0  
4  Germany     110.0   German     Not Rated     9950.0        NaN  
5      USA     100.0  English        Passed  2808000.0   379000.0  
6      USA      96.0  English        Passed        NaN  3950000.0  
7      USA      79.0  English       Unrated        NaN   800000.0  
8      USA      89.0  English       Unrated  2300000.0   439000.0  
9      USA      66.0  English      Approved        NaN   200000.0

In [27]:
# How many records are contained in films table
films.shape

(4968, 9)

total 4968 rows and 9 columns are contained.

---

## Filtering Rows

In [33]:
# Get all the details for all the films released in 2016
released_movies_2016 = films[films["release_year"] == 2016]
released_movies_2016.head(7)

id                            title  release_year country  duration  \
4820  4821              10 Cloverfield Lane        2016.0     USA     104.0   
4821  4822                         13 Hours        2016.0     USA     144.0   
4822  4823      A Beginner's Guide to Snuff        2016.0     USA      87.0   
4823  4824                          Airlift        2016.0   India     130.0   
4824  4825  Alice Through the Looking Glass        2016.0     USA     113.0   
4825  4826                        Allegiant        2016.0     USA     120.0   
4826  4827   Alleluia! The Devil's Carnival        2016.0     USA      97.0   

     language certification       gross       budget  
4820  English         PG-13  71897215.0   15000000.0  
4821  English             R  52822418.0   50000000.0  
4822  English           NaN         NaN          NaN  
4823    Hindi           NaN         NaN    4400000.0  
4824  English            PG  76846624.0  170000000.0  
4825  English         PG-13  66002193.0  110000000.0  
4826  English           NaN         NaN     500000.0

In [23]:
# Get the number of films released before 2016

filter_release_movie = films["release_year"] < 2016
released_before_2016 = films.loc[:, ["release_year"]] [filter_release_movie].value_counts(sort = False)
released_before_2016


release_year
1916.0            1
1920.0            1
1925.0            1
1927.0            1
1929.0            2
               ... 
2011.0          224
2012.0          220
2013.0          236
2014.0          252
2015.0          226
Length: 90, dtype: int64

In [13]:
# Get the title and release year of films released after 2000
fitler_movies_after_2000 = films["release_year"] > 2000
two_var = films.loc[:, ["title", "release_year"]]
released_after_2000 = two_var[fitler_movies_after_2000]
released_after_2000.head(7)

title  release_year
1508                    15 Minutes        2001.0
1509       3000 Miles to Graceland        2001.0
1510              A Beautiful Mind        2001.0
1511               A Knight's Tale        2001.0
1512  A.I. Artificial Intelligence        2001.0
1513                           Ali        2001.0
1514                   Alias Betty        2001.0

In [25]:
## Get all the details for all French language films
filter_french_lang = films["language"] == "French"
all_details = films[filter_french_lang]
all_details.head(7)


id                     title  release_year country  duration language  \
107  108          Une Femme Mariée        1964.0  France      94.0   French   
110  111            Pierrot le Fou        1965.0  France     110.0   French   
139  140       Mississippi Mermaid        1969.0  France     123.0   French   
422  423                    Subway        1985.0  France      98.0   French   
661  662             Les visiteurs        1993.0  France     107.0   French   
800  801  The Horseman on the Roof        1995.0  France     135.0   French   
915  916       When the Cat's Away        1996.0  France      91.0   French   

    certification      gross      budget  
107           NaN        NaN    120000.0  
110     Not Rated        NaN    300000.0  
139             R    26893.0   1600000.0  
422             R        NaN  17000000.0  
661             R   700000.0  50000000.0  
800             R  1877179.0         NaN  
915             R  1652472.0    300000.0

In [26]:
# Get new data

people = pd.read_csv("D:/documents/xlsx/people.csv")

In [27]:
people.head()

id                name   birthdate deathdate
0   1             50 Cent  1975-07-06       NaN
1   2  A. Michael Baldwin  1963-04-04       NaN
2   3       A. Raven Cruz         NaN       NaN
3   4        A.J. Buckley  1978-02-09       NaN
4   5        A.J. DeLucia         NaN       NaN

In [31]:
# Get the name and birth date of the person born on november 11th, 1974. Remember to use
# ISO date format ('1974-11-11')!
fitler_by_birthdate = people["birthdate"] == "1974-11-11"
get_names = people.loc[:, ["name", "birthdate"]] [fitler_by_birthdate]
get_names

name   birthdate
4791  Leonardo DiCaprio  1974-11-11

In [36]:
# Get the number of hindi language films

filter_hindi_lang = films["language"] == "Hindi"
number_of_hindi_language = films.loc[:, ["language"]] [filter_hindi_lang].value_counts()
number_of_hindi_language

language
Hindi       28
dtype: int64

In [38]:
# Get all details for all films with an R certification

r_certification_details = films["certification"] == 'R'
films[r_certification_details].head()

id                 title  release_year country  duration language  \
75    76                Psycho        1960.0     USA     108.0  English   
98    99  A Fistful of Dollars        1964.0   Italy      99.0  Italian   
133  134       Rosemary's Baby        1968.0     USA     136.0  English   
139  140   Mississippi Mermaid        1969.0  France     123.0   French   
144  145        The Wild Bunch        1969.0     USA     144.0  English   

    certification       gross     budget  
75              R  32000000.0   806947.0  
98              R   3500000.0   200000.0  
133             R         NaN  2300000.0  
139             R     26893.0  1600000.0  
144             R         NaN  6244087.0

In [45]:
# Get the title and release year for all Spanish language films released before 2000

filter_spanish_language_before_2000 = (films["language"] == "Spanish") & (films["release_year"] < 2000)
df1 = films.loc[:, ["title", "release_year"]] [filter_spanish_language_before_2000]
df1


title  release_year
621         El Mariachi        1992.0
1086  La otra conquista        1998.0
1128              Tango        1998.0

In [48]:
# Get all details for Spanish language films released after 2000

spanish_language = films["release_year"] < 2000
all_details2 = films[spanish_language]
all_details2.head()

id                                             title  release_year  \
0   1  Intolerance: Love's Struggle Throughout the Ages        1916.0   
1   2                    Over the Hill to the Poorhouse        1920.0   
2   3                                    The Big Parade        1925.0   
3   4                                        Metropolis        1927.0   
4   5                                     Pandora's Box        1929.0   

   country  duration language certification      gross     budget  
0      USA     123.0      NaN     Not Rated        NaN   385907.0  
1      USA     110.0      NaN           NaN  3000000.0   100000.0  
2      USA     151.0      NaN     Not Rated        NaN   245000.0  
3  Germany     145.0   German     Not Rated    26435.0  6000000.0  
4  Germany     110.0   German     Not Rated     9950.0        NaN

In [49]:
# Get all details for Spanish language films released after 2000, but before 2010
spanish_language2 = (films["release_year"] > 2000) & (films["release_year"] < 2010)
all_details3 = films[spanish_language2]
all_details3.head()

id                         title  release_year country  duration  \
1508  1509                    15 Minutes        2001.0     USA     120.0   
1509  1510       3000 Miles to Graceland        2001.0     USA     125.0   
1510  1511              A Beautiful Mind        2001.0     USA     135.0   
1511  1512               A Knight's Tale        2001.0     USA     144.0   
1512  1513  A.I. Artificial Intelligence        2001.0     USA     146.0   

     language certification        gross       budget  
1508  English             R   24375436.0   42000000.0  
1509  English             R   15738632.0   42000000.0  
1510  English         PG-13  170708996.0   58000000.0  
1511  English         PG-13   56083966.0   65000000.0  
1512  English         PG-13   78616689.0  100000000.0

In [51]:
# Get the title and release year for films released in the 90s
movies_in_ninties = films.loc[:, ["title", "release_year"]] [(films["release_year"] >= 1990) & (films["release_year"] < 2000)]
movies_in_ninties.head(10)

title  release_year
551                Arachnophobia        1990.0
552  Back to the Future Part III        1990.0
553               Child's Play 2        1990.0
554           Dances with Wolves        1990.0
555              Days of Thunder        1990.0
556                   Dick Tracy        1990.0
557                   Die Hard 2        1990.0
558          Edward Scissorhands        1990.0
559                   Flatliners        1990.0
560                        Ghost        1990.0

In [52]:
# the records to only include French or Spanish language films. show title and release year

two_language = films.loc[:, ["title", "release_year"]] [(films["language"] == "Spanish") | (films["language"] == "French")]
two_language


title  release_year
107                             Une Femme Mariée        1964.0
110                               Pierrot le Fou        1965.0
139                          Mississippi Mermaid        1969.0
422                                       Subway        1985.0
621                                  El Mariachi        1992.0
...                                          ...           ...
4630  Chiamatemi Francesco - Il Papa della gente        2015.0
4656                                   Evolution        2015.0
4799             They Will Have to Kill Us First        2015.0
4803                              Top Cat Begins        2015.0
4861                               Irreplaceable        2016.0

[112 rows x 2 columns]

In [61]:
# the records to only include French or Spanish language films. show title and release year
# Finally, restrict the query to only return films that took in more than $2M gross
df = films.loc[:, ["title", "release_year"]] [((films["language"] == "French") | (films["language"] == "Spanish")) & (films["gross"] > 2000000)]
df.head(7)

title  release_year
621                 El Mariachi        1992.0
1148             The Red Violin        1998.0
1344              Amores Perros        2000.0
1486  The Widow of Saint-Pierre        2000.0
1517                     Amélie        2001.0
1694          Y Tu Mamá También        2001.0
1700                    8 Women        2002.0